templates/example_cell_clustering.ipynb

```diff
@@ -11,7 +11,7 @@
    "cell_type": "markdown",
```

Comment:
<p>This step is giving me errors when testing. I think the problem is the discrepancy of how fovs is formatted, which depends on the way they are pulled from the MIBI directory at the beginning of the notebook. In the example dataset I'm working with, fovs are formatted like "Point6." See error message below. The problem seems to be at line 342 in create_c2pc_data, which I think expects the points to be formatted as "fov6." Perhaps adding a prefix input variable would be helpful here or somehow checking that it is consistent.</p><p> </p><pre class="ql-syntax" spellcheck="false">---------------------------------------------------------------------------
ValueError&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; Traceback (most recent call last)
&lt;ipython-input-17-db5c41c3f54c&gt; in &lt;module&gt;
&nbsp; &nbsp; &nbsp; 7&nbsp; &nbsp; &nbsp;weights_name=cell_weights_name,
&nbsp; &nbsp; &nbsp; 8&nbsp; &nbsp; &nbsp;cluster_col=cluster_col,
----&gt; 9&nbsp; &nbsp; &nbsp;num_passes=10
&nbsp; &nbsp; &nbsp;10 )


/usr/local/lib/python3.6/site-packages/ark/phenotyping/som_utils.py in train_cell_som(fovs, base_dir, pixel_consensus_dir, cell_table_name, cluster_counts_name, cluster_col, weights_name, xdim, ydim, lr_start, lr_end, num_passes, seed)
&nbsp; &nbsp; 943&nbsp; &nbsp; &nbsp;print("Counting the number of pixel SOM/meta cluster counts for each fov/cell pair")
&nbsp; &nbsp; 944&nbsp; &nbsp; &nbsp;cluster_counts = create_c2pc_data(
--&gt; 945&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;fovs, consensus_path, cell_table_path, cluster_col
&nbsp; &nbsp; 946&nbsp; &nbsp; &nbsp;)
&nbsp; &nbsp; 947&nbsp;


/usr/local/lib/python3.6/site-packages/ark/phenotyping/som_utils.py in create_c2pc_data(fovs, pixel_consensus_path, cell_table_path, cluster_col)
&nbsp; &nbsp; 340&nbsp;
&nbsp; &nbsp; 341&nbsp; &nbsp; &nbsp;# because current version of pandas doesn't support key-based sorting, need to do it this way
--&gt; 342&nbsp; &nbsp; &nbsp;cell_table['fov'] = cell_table['fov'].map(lambda x: x.replace('fov', '')).astype(int)
&nbsp; &nbsp; 343&nbsp; &nbsp; &nbsp;cell_table = cell_table.sort_values(by=['fov', 'segmentation_label']).reset_index(drop=True)
&nbsp; &nbsp; 344&nbsp; &nbsp; &nbsp;cell_table['fov'] = cell_table['fov'].map(lambda x: 'fov' + str(x))


/usr/local/lib/python3.6/site-packages/pandas/core/generic.py in astype(self, dtype, copy, errors, **kwargs)
&nbsp; &nbsp;5880&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# else, only a single dtype is given
&nbsp; &nbsp;5881&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;new_data = self._data.astype(
-&gt; 5882&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;dtype=dtype, copy=copy, errors=errors, **kwargs
&nbsp; &nbsp;5883&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;)
&nbsp; &nbsp;5884&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return self._constructor(new_data).__finalize__(self)


/usr/local/lib/python3.6/site-packages/pandas/core/internals/managers.py in astype(self, dtype, **kwargs)
&nbsp; &nbsp; 579&nbsp;
&nbsp; &nbsp; 580&nbsp; &nbsp; &nbsp;def astype(self, dtype, **kwargs):
--&gt; 581&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return self.apply("astype", dtype=dtype, **kwargs)
&nbsp; &nbsp; 582&nbsp;
&nbsp; &nbsp; 583&nbsp; &nbsp; &nbsp;def convert(self, **kwargs):


/usr/local/lib/python3.6/site-packages/pandas/core/internals/managers.py in apply(self, f, axes, filter, do_integrity_check, consolidate, **kwargs)
&nbsp; &nbsp; 436&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;kwargs[k] = obj.reindex(b_items, axis=axis, copy=align_copy)
&nbsp; &nbsp; 437&nbsp;
--&gt; 438&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;applied = getattr(b, f)(**kwargs)
&nbsp; &nbsp; 439&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;result_blocks = _extend_blocks(applied, result_blocks)
&nbsp; &nbsp; 440&nbsp;


/usr/local/lib/python3.6/site-packages/pandas/core/internals/blocks.py in astype(self, dtype, copy, errors, values, **kwargs)
&nbsp; &nbsp; 557&nbsp;
&nbsp; &nbsp; 558&nbsp; &nbsp; &nbsp;def astype(self, dtype, copy=False, errors="raise", values=None, **kwargs):
--&gt; 559&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return self._astype(dtype, copy=copy, errors=errors, values=values, **kwargs)
&nbsp; &nbsp; 560&nbsp;
&nbsp; &nbsp; 561&nbsp; &nbsp; &nbsp;def _astype(self, dtype, copy=False, errors="raise", values=None, **kwargs):


/usr/local/lib/python3.6/site-packages/pandas/core/internals/blocks.py in _astype(self, dtype, copy, errors, values, **kwargs)
&nbsp; &nbsp; 641&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# _astype_nansafe works fine with 1-d only
&nbsp; &nbsp; 642&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;vals1d = values.ravel()
--&gt; 643&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;values = astype_nansafe(vals1d, dtype, copy=True, **kwargs)
&nbsp; &nbsp; 644&nbsp;
&nbsp; &nbsp; 645&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# TODO(extension)


/usr/local/lib/python3.6/site-packages/pandas/core/dtypes/cast.py in astype_nansafe(arr, dtype, copy, skipna)
&nbsp; &nbsp; 705&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# work around NumPy brokenness, #1987
&nbsp; &nbsp; 706&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;if np.issubdtype(dtype.type, np.integer):
--&gt; 707&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;return lib.astype_intsafe(arr.ravel(), dtype).reshape(arr.shape)
&nbsp; &nbsp; 708&nbsp;
&nbsp; &nbsp; 709&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;# if we have a datetime/timedelta array of objects


pandas/_libs/lib.pyx in pandas._libs.lib.astype_intsafe()


ValueError: invalid literal for int() with base 10: 'Point6'
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/angelolab/ark-analysis/pull/460/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Reply:
We shouldn't assume that FOVs are named in a particular way. People will be using this with non-MIBI data as well.

Reply:
<p>Ah good point, let me patch this up.</p>

Reply:
<p>Actually, nvm. The sorting was only to confirm the results matched up with Candace's dataset (because the order the cells mattered for training). We can remove the sorting now.</p>